arbnode/batch_poster.go

```diff
@@ -1238,7 +1241,24 @@ func (b *BatchPoster) maybePostSequencerBatch(ctx context.Context) (bool, error)
 		b.building.msgCount++
 	}
 
-	if !forcePostBatch || !b.building.haveUsefulMessage {
+	var disablePosting bool
+	firstMsgTimeStamp := firstMsg.Message.Header.Timestamp
+	firstMsgBlockNumber := firstMsg.Message.Header.BlockNumber
+	batchNearL1BoundMinTimestamp := firstMsgTimeStamp >= l1BoundMinTimestamp && firstMsgTimeStamp <= l1BoundMinTimestamp+uint64(config.ReorgResistanceMargin/time.Second)
+	batchNearL1BoundMinBlockNumber := firstMsgBlockNumber >= l1BoundMinBlockNumber && firstMsgBlockNumber <= l1BoundMinBlockNumber+uint64(config.ReorgResistanceMargin/ethPosBlockTime)
+	if config.ReorgResistanceMargin > 0 && (batchNearL1BoundMinTimestamp || batchNearL1BoundMinBlockNumber) {
+		log.Error(
+			"Disabling batch posting due to batch being within reorg resistance margin from layer 1 minimum block or timestamp bounds",
+			"reorgResistanceMargin", config.ReorgResistanceMargin,
+			"firstMsgTimeStamp", firstMsgTimeStamp,
+			"l1BoundMinTimestamp", l1BoundMinTimestamp,
+			"firstMsgBlockNumber", firstMsgBlockNumber,
+			"l1BoundMinBlockNumber", l1BoundMinBlockNumber,
+		)
+		disablePosting = true
```

Comment:
I think we should just return an error here. We should keep the log line above to provide a bunch of details, but erroring here will result in "error posting batch" which teams might have log based alerting for.